pkg/git/git.go

```diff
@@ -182,7 +198,10 @@ func TempClone(gitURL, hash string, useAuth bool) (repoDir string, err error) {
 
 	var auth transport.AuthMethod
 	if useAuth {
-		auth = GetGitAuth()
+		auth, err = GetGitAuth(gitURL)
```

Comment:
Under what conditions does `TempClone` need authenticated access to GitHub? It seems like read-only usage.

Reply:
Looks like the only occurrences of `TempClone`  are called from [advisory diff](https://github.com/wolfi-dev/wolfictl/blob/922b5a2bb4e1915b7f4d05727bdd1f2d1c1a9585/pkg/cli/advisory_diff.go#L42) and [advisory validate](https://github.com/wolfi-dev/wolfictl/blob/922b5a2bb4e1915b7f4d05727bdd1f2d1c1a9585/pkg/cli/advisory_validate.go#L145).  These could be private advisory repos AFAIK which would require auth.  My understanding after looking at the code is that the `Temp` part relates to cloning into a temp dir.  

@luhring can confirm,

Reply:
That's correct!